docs/docs/configuration.md

```diff
@@ -143,6 +143,23 @@ The properties can be manually constructed or passed in from a compute engine li
 Spark uses its session properties as catalog properties, see more details in the [Spark configuration](spark-configuration.md#catalog-configuration) section.
 Flink passes in catalog properties through `CREATE CATALOG` statement, see more details in the [Flink](flink.md#adding-catalogs) section.
 
+### Catalog REST auth properties
+
+The following catalog properties configure authentication for the REST catalog.
+They support Basic, OAuth2, and SigV4 authentication, in addition to the default none.
+
+| Property                          | Default            | Description                                            |
+| --------------------------------- | ------------------ | ------------------------------------------------------ |
+| rest.auth.type                    | none               | Authentication mechanism for REST catalog access. Supported values: `none`, `basic`, `oauth2`, `sigv4`. |
+| rest.auth.basic.username          | null               | Username for Basic authentication. Required if `rest.auth.type` = `basic`. |
+| rest.auth.basic.password          | null               | Password for Basic authentication. Required if `rest.auth.type` is `basic`. |
```

Comment:
How about extracting tables for each auth type?

Reply:
There are also some shared properties, so, we might end up with 3-4 small tables. I am open to suggestion if you think that'll be a cleaner approach. Please let me know what you think.